aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -24,6 +24,48 @@
 #include <numeric>
 
 namespace at {
+
+namespace meta {
+
+TORCH_META_FUNC(addmm)(const Tensor &self, const Tensor &mat1, const Tensor &mat2,
+                       const Scalar& beta, const Scalar& alpha) {
+  TORCH_CHECK(mat1.dim() == 2, "mat1 must be a matrix, got ",
+              mat1.dim(), "-D tensor");
+  TORCH_CHECK(mat2.dim() == 2, "mat2 must be a matrix, got ",
+              mat2.dim(), "-D tensor");
+
+  const auto self_sizes = self.sizes();
+  auto m1_sizes = mat1.sizes();
+  auto m2_sizes = mat2.sizes();
+
+  if (self.numel() != 0) {
+    // if number of elements are zero, only then can the input tensor can be resized.
+    TORCH_CHECK((self.dim() == 2), "input matrix must be a matrix, got ",
+                self.dim(), "-D tensor");
+    TORCH_CHECK(((self_sizes[0] == m1_sizes[0]) && (self_sizes[1] == m2_sizes[1])),
+                "The input matrix must have the same size as the product of mat1 & mat2");
+  }
+
+  TORCH_CHECK(
+      m1_sizes[1] == m2_sizes[0], "mat1 and mat2 shapes cannot be multiplied (",
+      m1_sizes[0], "x", m1_sizes[1], " and ", m2_sizes[0], "x", m2_sizes[1], ")");
+
+  auto names = at::namedinference::propagate_names_for_addmm(mat1, mat2, self);
+  const int64_t size_[2] = {mat1.sizes().data()[0], mat2.sizes().data()[1]};
+  set_output(0, IntArrayRef(size_, 2), {}, self.options(), names);
+  auto result = maybe_get_output(0);
+  // This check can fire for inplace op only, for all other versions result is
```

Comment:
This check I copy-pasted from the `addmv` PR would help with the inplace variant, but after the computation.